Prototype

chrismytton opened the pull request:
This uses code originally developed in https://github.com/everypolitician-scrapers/guernsey-2016/pull/1 and https://github.com/everypolitician-scrapers/mongolia-khurai-wp-multiple-terms/pull/7 to give us a first version of the gem. Those two pull requests have also been updated to use the code in this pull request to demonstrate it working in a real world setting.

# Notes to reviewer

I would like to strap some automated tests onto this prototype ASAP, but I think there's value in releasing this version as `v0.1.0` and then iterating on the library from there.

# Notes to merger

- [x] This should not be merged until `scraped_page_archive` v0.5.0 has been released. See https://github.com/everypolitician/scraped_page_archive/pull/55.

chrismytton committed:
Add runtime dependencies to gemspec

These are the gems that we need to glue together to make the ScrapedPage
class.

chrismytton committed:
Initial implementation of ScrapedPage class

This implementation was ported over from the prototypes that I did in
the `guernsey-2016` and `mongolia-khurai-wp-multiple-terms` scrapers.

At the moment I'm just dumping it all in one big file, but it would be
good to split this into separate files sooner rather than later.

chrismytton committed:
Require at least version 0.5.0 of scraped_page_archive

Previous versions of scraped_page_archive work slightly differently, so
we need to ensure we're working with the latest version.

Note at time of commit this version hadn't actually been released, but I
installed it into my local gems by running the following command in my
local clone:

    bundle exec rake install:local

chrismytton commented:
The Travis tests are failing because it needs https://github.com/everypolitician/scraped_page_archive/pull/55 to be merged and v0.5.0 of `scraped_page_archive` to be released first.

chrismytton commented:
@tmtmtmtm :eyes:

tmtmtmtm commented on lib/scraped_page.rb:
```diff
@@ -1,5 +1,74 @@
+      private
+
+      def log(message)
+        warn "[#{self.class}] #{message}"
```
I think I'd prefer this to be turned off by default, considering the problems that Morph has with lots of console output. Perhaps even for the first version we could have it only output warnings if an ENV var is set whilst we work out what a better approach should be?

chrismytton replied:
Ah yeah, this was handy while I was developing, but yep, no need for it to be on by default. I've added a `VERBOSE` environment variable to enable logging in https://github.com/everypolitician/scraped_page/pull/7/commits/6666c84ad12f863a2c142812499d92c2753b09b8.

tmtmtmtm reviewed (changes requested):
Mostly looking good, yay! As well as one tiny query about default logging, I think this one also falls into the territory of really wanting https://github.com/bbatsov/ruby-style-guide#one-class-per-file

chrismytton committed:
Move classes into their own files

chrismytton committed:
Only log if VERBOSE=1 in environment

Outputting lots of log lines can be troublesome for Morph.

It's also more in line with the "Rule of Silence" from the UNIX
Philosophy.

chrismytton commented:
> I think this one also falls into the territory of really wanting https://github.com/bbatsov/ruby-style-guide#one-class-per-file

Yep, it was useful having it in one file when I was copying it between scrapers to test it, but no need for that now its in the gem. I've split them up in https://github.com/everypolitician/scraped_page/pull/7/commits/39f2de9f021d123361cfe89cdff2bd7ef5a02cfe.

chrismytton commented:
@tmtmtmtm 👀
